extensions/sql-migration/src/dialog/skuRecommendationResults/importPerformanceDataDialog.ts

```diff
@@ -0,0 +1,279 @@
+				'',
+				'ImportPerformanceDataDialog',
+				350,
+				'callout',
+				'below',
+				false,
+				false,
+				<azdata.window.IDialogProperties>{
+					height: 51,
+					width: 0,
+					xPos: 0,
+					yPos: 0
+				}
+			);
+
+			const promise = this.initializeDialog(this.dialog);
+			azdata.window.openDialog(this.dialog);
+			await promise;
+		}
+	}
+
+	private createOpenExistingFolderContainer(_view: azdata.ModelView): azdata.FlexContainer {
+		const container = _view.modelBuilder.flexContainer()
+			.withProps({
+				CSSStyles: {
+					'flex-direction': 'column',
+					'margin-top': '10px'
+				}
+			})
+			.component();
+
+		const instructions = _view.modelBuilder.text().withProps({
+			value: constants.IMPORT_PERFORMANCE_DATA_DIALOG_HELPER_MESSAGE,
+			CSSStyles: {
+				'font-size': '13px',
+				'line-height': '18px',
+				'font-weight': '600',
+			}
+		}).component();
+
+		const selectFolderContainer = _view.modelBuilder.flexContainer()
+			.withProps({
+				CSSStyles: {
+					'flex-direction': 'row',
+					'align-items': 'center',
+					'marign-right': '30px'
+				}
+			})
+			.component();
+
+		this._openExistingFolderInput = _view.modelBuilder.inputBox().withProps({
+			placeHolder: constants.IMPORT_PERFORMANCE_DATA_DIALOG_OPEN_FILE,
+			readOnly: true,
+			height: 24,
+			width: 222,
+			CSSStyles: {
+				'font-size': '13px',
+				'line-height': '18px',
+				'font-weight': '400',
+				'margin': '0px',
+			},
+		}).component();
+		this._disposables.push(
+			this._openExistingFolderInput.onTextChanged(async (value) => {
+				if (value) {
+					this.migrationStateModel._skuRecommendationPerformanceLocation = value.trim();
+					this.dialog!.okButton.enabled = true;
+					this._importButton.enabled = true;
+				}
+			}));
+
+		const openButton = _view.modelBuilder.button()
+			.withProps({
+				iconHeight: 24,
+				iconWidth: 24,
+				iconPath: IconPathHelper.openFolder,
+				CSSStyles: {
+					'margin': '0',
+					'padding': '0',
+				}
+			}).component();
+		this._disposables.push(
+			openButton.onDidClick(
+				async (e) => this._openExistingFolderInput.value = await utils.promptUserForFolder()));
+
+		selectFolderContainer.addItems([
+			this._openExistingFolderInput,
+			openButton]);
+		container.addItems([
+			instructions,
+			selectFolderContainer]);
+
+		return container;
+	}
+
+	protected async execute() {
+		this._isOpen = false;
+		this.migrationStateModel._skuRecommendationPerformanceDataSource = this._performanceDataSource;
+
+		const serverName = (await getSourceConnectionProfile()).serverName;
+		const errors: string[] = [];
+		try {
+			// TODO - ADD the card loading.
+			// await this.skuRecommendationPage.startCardLoading();
+			await this.migrationStateModel.getSkuRecommendations();
+
+			const skuRecommendationError = this.migrationStateModel._skuRecommendationResults?.recommendationError;
+			if (skuRecommendationError) {
+				errors.push(`message: ${skuRecommendationError.message}`);
```

Comment:
be sure to add all user strings to a resource string for localization.

Reply:
done